Animal prediction

Viidas96 opened the pull request:
Adding animal data in csv format file.
Created elephant_regression_model
Created RegressionClass

Viidas96 committed:
Created Json file for elphant data

This json file contains 280 000+ records of elephant data

Viidas96 committed:
Organisng Data

Created class that will be our model
Also created two functions that read and organise the data

BCybercell committed:
Created animal prediction model

Viidas96 committed:
Update animal-prediction-model.ts

Made Changes to inccorect functions

Viidas96 committed:
Made Changes To ML

ML is up and running. The outputs and predictions are not correct though.
Either problem with model or problem with input data

Viidas96 committed:
Merge remote-tracking branch 'origin/development' into animal-prediction

Viidas96 committed:
Merged Development into here

Merged

Viidas96 committed:
Animal Prediction Service

Service

Viidas96 committed:
Update animal-prediction.service.ts

Load Model updates

Viidas96 committed:
Added Predict function

Changed file name to Regression model
Added predict function

Viidas96 committed:
Merge branch 'development' into animal-prediction

Viidas96 committed:
Regression Model Functions

Regression Model functions tested and error free. Model now saves and loads.Trains and is able to predict.
Future update is getching model details

EvansMatthew97 commented on server/src/providers/regression-model.service.ts:
```diff
@@ -0,0 +1,75 @@
+    private logsEnabled: boolean;
+    private inputs: any;
+    private outputs: any;
+    constructor(logsEnabled= false) {
```
```suggestion
    constructor(private logsEnabled = false) {
```

EvansMatthew97 commented on server/src/providers/regression-model.service.ts:
```diff
@@ -0,0 +1,75 @@
+    private inputs: any;
+    private outputs: any;
+    constructor(logsEnabled= false) {
+        this.logsEnabled = logsEnabled;
```
```suggestion
```

EvansMatthew97 commented on server/src/providers/regression-model.service.ts:
```diff
@@ -0,0 +1,75 @@
+    }
+
+    // Prediction of the model
+    public predict(inputs) {
```
I don't think this needs to be wrapped in a try/catch since it will throw the error anyways

EvansMatthew97 commented on server/src/providers/regression-model.service.ts:
```diff
@@ -0,0 +1,75 @@
+    }
+
+    // Saves a model to HDD 
+    public saveModel(modelName) {
```
Try catch unnecessary again

EvansMatthew97 commented on server/src/providers/regression-model.service.ts:
```diff
@@ -0,0 +1,75 @@
+    // Saves a model to HDD 
+    public saveModel(modelName) {
+        try {
+        const jsonModel = JSON.stringify(this.model.toJSON());
```
Make sure that this.model.toJSON() doesn't return a JSON string already

EvansMatthew97 commented on server/src/providers/regression-model.service.ts:
```diff
@@ -0,0 +1,75 @@
+    public saveModel(modelName) {
+        try {
+        const jsonModel = JSON.stringify(this.model.toJSON());
+        fs.writeFileSync('ai_models/' + modelName + '.json', jsonModel);
```
```suggestion
        fs.writeFileSync(`'ai_models/${modelName}.json`, jsonModel);
```

Maybe abstract 'ai_models' into a constant at the top of the file so it can be changed easily.

EvansMatthew97 commented on server/src/providers/regression-model.service.ts:
```diff
@@ -0,0 +1,75 @@
+    }
+
+    // Loads a model from storage
+    public loadModel(modelName) {
```
Unnecessary try catch

EvansMatthew97 commented on server/src/providers/regression-model.service.ts:
```diff
@@ -0,0 +1,75 @@
+    // Loads a model from storage
+    public loadModel(modelName) {
+        try {
+            if(fs.existsSync('ai_models/' + modelName + '.json'))
```
```suggestion
            if (fs.existsSync(`ai_models/${modelName}.json`))
```

EvansMatthew97 commented on server/src/providers/regression-model.service.ts:
```diff
@@ -0,0 +1,75 @@
+    public loadModel(modelName) {
+        try {
+            if(fs.existsSync('ai_models/' + modelName + '.json'))
+            {
```
Put brackets on the same line as the if statement:

```
if (something {
  // code block
}
```

EvansMatthew97 commented on server/src/providers/regression-model.service.ts:
```diff
@@ -0,0 +1,75 @@
+    // Logs messages
+    private log(message)
+    {
+        if (this.logsEnabled === true) {
```
```suggestion
        if (this.logsEnabled) {
```

EvansMatthew97 commented on server/src/providers/regression-model.service.ts:
```diff
@@ -0,0 +1,75 @@
+    // Prediction of the model
+    public predict(inputs) {
+        try {
+            const predictions = [];
```
```suggestion
            const predictions = inputs.map(input => this.model.predict(input));
```

Then you can remove the .forEach code block

EvansMatthew97 reviewed (changes requested):
It looks like you aren't getting the animal points from the database anywhere?

It should read those points and train based off of that.

Viidas96 committed:
Full Functional

Fully functional model
Model shows client side for demo
Logic checks need to happen tomorrow to validate outputs

Viidas96 committed:
Delete animal_location.csv

Delete this file because not needed

Viidas96 committed:
Update routes.page.ts

Neaten code changes

Viidas96 committed:
Logic Changes

Client Demo of regression looks more accurate with minor logic changes
